src/main/java/seedu/address/logic/parser/cookbook/CookbookCommandParser.java

```diff
@@ -9,12 +19,38 @@
  */
 public class CookbookCommandParser implements Parser<CookbookCommand> {
 
+    private static final Pattern COOKBOOK_COMMAND_ARGUMENT_FORMAT = Pattern
+            .compile("(?<commandWord>\\S+) (?<index>\\S+)");
+    private static final Pattern COOKBOOK_ADD_COMMAND_ARGUMENT_FORMAT = Pattern
+            .compile("(?<commandWord>\\S+) (?<arguments>.*)");
```

Comment:
We can just use one like `private static final Pattern COOKBOOK_COMMAND_ARGUMENT_FORMAT = Pattern.compile("(?<commandWord>\\S+) (?<arguments>.*)");`

Then pass the arguments to each parser to parse. Can refer to the original AddressBookParser, no matter whether the arguments are prefixed or are indices, they are passed directly to the parsers.